### PR TITLE
Switches to cudnn-frontend pypi package

### PR DESCRIPTION
**Pull request by vedaanta**
<details>
  <summary><b>Before submitting</b></summary>

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

</details>

## What does this PR do?
cuDNN frontend wheels are now available from pypi. [Link](https://pypi.org/project/nvidia-cudnn-frontend/)

This PR uses pypi wheels for cudnn frontend instead of building them from source.

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

File: dockers/ubuntu-cuda/Dockerfile [hunk 1]
```diff
@@ -20,7 +20,7 @@ ARG IMAGE_TYPE="devel"
 FROM nvidia/cuda:${CUDA_VERSION}-${IMAGE_TYPE}-ubuntu${UBUNTU_VERSION}
 
 ARG CUDNN_VERSION="9.1.0.70"
-ARG CUDNN_FRONTEND_CHECKOUT="v1.4.0"
+ARG CUDNN_FRONTEND_CHECKOUT="1.4.0"
```
Comment (Borda) [inline dockers/ubuntu-cuda/Dockerfile:23]:
```suggestion
ARG CUDNN_FRONTEND_VERSION="1.4.0"
```
Since it is version now, not building from git/source

Comment (vedaanta) [reply dockers/ubuntu-cuda/Dockerfile]: done!

File: .azure/docker-build.yml [hunk 2]
```diff
@@ -40,9 +40,9 @@ jobs:
       #maxParallel: "3"
       matrix:
         # CUDA 12.1
-        "cuda 12.1 | torch 2.3 | cudnn FE v1.3":
+        "cuda 12.1 | torch 2.3 | cudnn FE v1.4":
           { CUDA_VERSION: "12.1.1", TORCH_VERSION: "2.3.0", TRITON_VERSION: "2.3.0", CUDNN_FRONTEND: "1.4.0" }
```
Comment (Borda) [inline .azure/docker-build.yml:44]:
```suggestion
          { CUDA_VERSION: "12.1.1", TORCH_VERSION: "2.3.0", TRITON_VERSION: "2.3.0", CUDNN_FRONTEND_VERSION: "1.4.0" }
```

File: .azure/docker-build.yml [hunk 3]
```diff
@@ -40,9 +40,9 @@ jobs:
...
-        "cuda 12.1 | torch 2.4 /nightly | cudnn FE v1.3":
+        "cuda 12.1 | torch 2.4 /nightly | cudnn FE v1.4":
           { CUDA_VERSION: "12.1.1", TORCH_VERSION: "main", TORCH_INSTALL: "source", CUDNN_FRONTEND: "1.4.0" }
```
Comment (Borda) [inline .azure/docker-build.yml:46]:
```suggestion
          { CUDA_VERSION: "12.1.1", TORCH_VERSION: "main", TORCH_INSTALL: "source", CUDNN_FRONTEND_VERSION: "1.4.0" }
```

**Review by Borda — APPROVED** [review]

**Comment by vedaanta** [comment]
@t-vi @carmocca can you please approve and merge? Thanks!

**Review by carmocca — APPROVED** [review]

**Comment by carmocca** [comment]
Done 😄

**Pull request merged**